From 93714624efa46263985a3183722836de4022304a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ji=C5=99=C3=AD=20Klime=C5=A1?= <jklimes@redhat.com>
Date: Tue, 24 Feb 2015 12:34:53 +0100
Subject: [PATCH] editor: disable "Ask for this password..." for system
 connections (rh #1173245)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

System connections cannot ask for password, so they have to have password
stored (IEEE_8021X_PASSWORD).

https://bugzilla.redhat.com/show_bug.cgi?id=1173245
Signed-off-by: Jiří Klimeš <jklimes@redhat.com>
---
 .../src/wireless-security/eap-method-simple.c                    | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/network-manager-applet-0.8.1/src/wireless-security/eap-method-simple.c b/network-manager-applet-0.8.1/src/wireless-security/eap-method-simple.c
index 34e8aa5..4d95059 100644
--- a/network-manager-applet-0.8.1/src/wireless-security/eap-method-simple.c
+++ b/network-manager-applet-0.8.1/src/wireless-security/eap-method-simple.c
@@ -82,6 +82,7 @@ fill_connection (EAPMethod *parent, NMConnection *connection)
 	NMSetting8021x *s_8021x;
 	NMSettingConnection *s_con;
 	gboolean always_ask;
+	NMConnectionScope scope;
 
 	s_8021x = NM_SETTING_802_1X (nm_connection_get_setting (connection, NM_TYPE_SETTING_802_1X));
 	g_assert (s_8021x);
@@ -112,6 +113,14 @@ fill_connection (EAPMethod *parent, NMConnection *connection)
 
 	g_object_set (s_8021x, NM_SETTING_802_1X_IDENTITY, gtk_entry_get_text (method->username_entry), NULL);
 
+	/* Always-ask password is not allowed for system connections. There would
+	 * be nobody to ask user for the password.
+ 	 */
+	scope = nm_connection_get_scope (connection);
+	if (scope == NM_CONNECTION_SCOPE_SYSTEM)
+		gtk_toggle_button_set_active (method->always_ask, FALSE);
+	gtk_widget_set_sensitive (GTK_WIDGET (method->always_ask), scope == NM_CONNECTION_SCOPE_USER);
+
 	/* Save the password always ask setting */
 	always_ask = gtk_toggle_button_get_active (method->always_ask);
 
-- 
2.1.0

